### ISSUE TITLE
[Upstream] Deb/RPM packaging: issues with compressed/split debug info

### ISSUE THREAD

**Issue by LebedevRI** [darktable-org/darktable#16322]
In https://github.com/darktable-org/rawspeed/pull/658,
i've changed rawspeed build system to tell compiler/linker
to compress debug info, and split it into separate files.
That resulted in enormous disk space savings for me.

But now that it got propagated into dt master,
https://build.opensuse.org/package/show/graphics:darktable:master/darktable
builds gone red.

i've backtracked those changes now, so it should go green for now,
and stable release is not going to be borked, but this is not a long-term solution.
I'm not quite sure how the packaging recepies need to be adjusted
so that they are happy with split/compressed debug info.

CC @cryptomilk @bremner @TurboGit

**Comment by darix** [comment]
those packaging tools already strip out the debuginfo later anyway and ship them into different packages. you do not have to do it manually in the build system.

**Comment by LebedevRI** [comment]
The thing is that i **DO** want to do it in my build system, because
it reduces the size of my build directory from 800 megabytes down to 200 megabytes.

**Comment by darix** [comment]
then hide it behind a cmake option and dont break stuff with your assumptions without need.

**Comment by LebedevRI** [comment]
Ah yes, feels like the old-school forums :)
"> How do i do XYZ specifically? 
< Who even does that, RTFM!"

**Comment by darix** [comment]
it is not the first time that you break stuff to achieve for yourself without caring about the wider consequences. but keep up your attitude.

**Comment by LebedevRI** [comment]
I don't appreciate this kind of commentary. If you have nothing to add, don't add anything.

**Comment by cryptomilk** [comment]
All distributions do not support if you modify objects in any form: striping partially, splitting compressing etc. They just set the flags they want and then extract the debuginfo. So the default build should assume the defaults. If you have a `ENABLE_DEVELOPER` option of some kind, the best is to hide it behind this as distros wont turn that on.

On the project I'm working on we have something like this which enables really pedantic compiler options and things you do not want in distributions that they fail on that because they update to a newer compiler version.

If you don't have a `ENABLE_DEVELOPER` yet, maybe add one for you and other developers :-)

**Comment by LebedevRI** [comment]
...

Ok, let me try again :)

I'm afraid this sounds like you are just saying the usual thing of
"the distros are so special, just keep your head down not to anger them".

I have to ask, have you actually assessed what the failures i mentioned are?
Have you read https://fedoraproject.org/wiki/Features/DwarfCompressor, 
https://sourceware.org/pipermail/dwz/2021q1/000886.html,
https://sourceware.org/bugzilla/show_bug.cgi?id=24756 ?

For example, the issue at hand does not appear to happen on openSuse, only Fedora / debian/ubuntu.

**Comment by ptilopteri** [comment]
* Roman Lebedev ***@***.***> [02-16-24 11:41]:
> ...
> 
> Ok, let me try again :)
> 
> I'm afraid this sounds like you are just saying the usual thing of
> "the distros are so special, just keep your head down not to anger them".
> 
> I have to ask, have you actually assessed what the failures i mentioned are?
> Have you read https://fedoraproject.org/wiki/Features/DwarfCompressor, 
> https://sourceware.org/pipermail/dwz/2021q1/000886.html,
> https://sourceware.org/bugzilla/show_bug.cgi?id=24756 ?
> 
> For example, the issue at hand does not appear to happen on openSuse, only Fedora / debian/ubuntu.
> Message ID: ***@***.***>


odd that, opensuse has failed since
darktable-4.7.0~git577.1620befa-11344.1.x86_64.rpm
on
https://download.opensuse.org/repositories/graphics:/darktable:/master/openSUSE_Tumbleweed/x86_64/
(14Feb24)

warning: %post(polkit-default-privs-1550+20240207.d833f4b-1.1.noarch)
scriptlet failed, exit status 2



-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
http://en.opensuse.org    openSUSE Community Member    facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by LebedevRI** [comment]
I'm talking about https://build.opensuse.org/package/show/graphics:darktable:master/rawspeed:clang / https://build.opensuse.org/package/show/graphics:darktable:master/rawspeed:gcc,
not https://build.opensuse.org/package/show/graphics:darktable:master/darktable
which is now failing for entirely unrelated reasons.

**Comment by LebedevRI** [comment]
Ok, `graphics:darktable:master/darktable` was affected by several issues in short succession (failure to find openexr package, exif.cc patch needed updating, plus the subj issue) so i've missed one more failure there:
This suse-specific thing:
```
```

**Comment by LebedevRI** [comment]
Ok, `graphics:darktable:master/darktable` was affected by several issues in short succession (failure to find openexr package, exif.cc patch needed updating, plus the subj issue) so i've missed one more failure there:
This suse-specific thing:
```
[  125s] + find . -type f -name CMakeLists.txt -exec sed -i -re '/^[[:blank:]]*[sS][eE][tT][[:blank:]]*\([[:blank:]]*(CMAKE_BUILD_TYPE|CMAKE_COLOR_MAKEFILE|CMAKE_INSTALL_PREFIX|CMAKE_VERBOSE_MAKEFILE).*\)/{s/^/#IGNORE /}' '{}' +
```
breaks
https://github.com/darktable-org/darktable/blob/26440d6094da1d27ee33890333794aa2e807ec11/src/external/CMakeLists.txt#L17-L22

**Comment by LebedevRI** [comment]
> odd that, opensuse has failed since darktable-4.7.0~git577.1620befa-11344.1.x86_64.rpm on https://download.opensuse.org/repositories/graphics:/darktable:/master/openSUSE_Tumbleweed/x86_64/ (14Feb24) warning: %post(polkit-default-privs-1550+20240207.d833f4b-1.1.noarch) scriptlet failed, exit status 2
> […](#)
> -- (paka)Patrick Shanahan Plainfield, Indiana, USA @ptilopteri http://en.opensuse.org openSUSE Community Member facebook/ptilopteri Photos: http://wahoo.no-ip.org/piwigo paka @ IRCnet oftc

FWIW, fixed.

**Comment by ptilopteri** [comment]
* Roman Lebedev ***@***.***> [02-16-24 23:05]:
> > odd that, opensuse has failed since darktable-4.7.0~git577.1620befa-11344.1.x86_64.rpm on https://download.opensuse.org/repositories/graphics:/darktable:/master/openSUSE_Tumbleweed/x86_64/ (14Feb24) warning: %post(polkit-default-privs-1550+20240207.d833f4b-1.1.noarch) scriptlet failed, exit status 2
> > […](#)
> 
> FWIW, fixed.
> Message ID: ***@***.***>


tks, openSUSE Tumbleweed build also passes now  :)


-- 
(paka)Patrick Shanahan       Plainfield, Indiana, USA          @ptilopteri
http://en.opensuse.org    openSUSE Community Member    facebook/ptilopteri
Photos: http://wahoo.no-ip.org/piwigo                   paka @ IRCnet oftc

**Comment by bremner** [comment]
"github-actions[bot]" ***@***.***> writes:

> This issue has been marked as stale due to inactivity for the last 60 days. It will be automatically closed in 300 days if no update occurs. Please check if the master branch has fixed it and report again or close the issue.
>
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/darktable-org/darktable/issues/16322#issuecomment-2062763571
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>

never surrender to the stalebot menace

**Comment by bremner** [comment]
"github-actions[bot]" ***@***.***> writes:

> Closed #16322 as not planned.
>
> -- 
> Reply to this email directly or view it on GitHub:
> https://github.com/darktable-org/darktable/issues/16322#event-17258185809
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>
Every time a stalebot closes a bug, I am less motivated to report
further bugs to that project. Just FYI.

**Comment by bremner** [comment]
wpferguson ***@***.***> writes:

> wpferguson left a comment (darktable-org/darktable#16322)
>
>> FWIW, fixed.
>
> Is it not fixed?

The last 3 messages from the bot did not suggest so. I did not re-read the other 19
messages the the thread.

**Comment by LebedevRI** [comment]
I'm reasonably sure the correct fix is to enable split+compressed debug info and skip/disable dwz,
and end up with NET decrease of dbgsym filesize.
